redhat/rdma-core.spec

```diff
@@ -346,7 +351,11 @@ rm -f %{buildroot}%{_sysconfdir}/rdma/modules/rdma.conf
 install -D -m0644 redhat/rdma.conf %{buildroot}%{_sysconfdir}/rdma/modules/rdma.conf
 
 # ibacm
+%if 0%{?fedora} >= 33
+%{buildroot}%{_bindir}/ib_acme -D . -O
+%else
 bin/ib_acme -D . -O
```

Comment:
Why can't we use buildroot everywhere? This looks kind of weird too, IIRC ib_acme writes a file, so under the 'out of source builds' concept it should be something more like

```
(cd %{buildroot} &&
 %{_bindir}/ib_acme -D . -O &&
 install -D -m0644 ibacm_opts.cfg %{buildroot}%{_sysconfdir}/rdma/)
```

Reply:
I will replace it with something like this:

```
(cd %{buildroot} &&
 ./%{_bindir}/ib_acme -D ./%{_sysconfdir}/rdma/ -O &&
 chmod 0644 ./%{_sysconfdir}/rdma/ibacm_opts.cfg)
```

Reply:
Can't change directory, the rpmbuild tool 'mock' is unhappy about this.

```
+ cd /builddir/build/BUILDROOT/rdma-core-33.0-1.el8.x86_64
+ .//usr/bin/ib_acme -D .//etc/rdma/ -O
.//usr/bin/ib_acme: error while loading shared libraries: libibverbs.so.1: cannot open shared object file: No such file or directory
```

Reply:
Oh, it can't be _bindir, it has to be ./bin/ib_acme  - it needs the compiled but not yet installed version to get the rpath to resolve the yet to be installed library.

Reply:
It seems it does not works for epel-8.
```
+ cd /builddir/build/BUILDROOT/rdma-core-33.0-1.el8.x86_64//usr
+ ./bin/ib_acme -D /builddir/build/BUILDROOT/rdma-core-33.0-1.el8.x86_64//etc/rdma/ -O
./bin/ib_acme: error while loading shared libraries: libibverbs.so.1: cannot open shared object file: No such file or directory
```

Reply:
That is the wrong path, //usr is the installation direction, you need to cd to the directory that holds the .o files

Reply:
```
(if [ -d %{__cmake_builddir} ]; then cd %{__cmake_builddir}; fi
 ./bin/ib_acme -D . -O &&
 install -D -m0644 ibacm_opts.cfg %{buildroot}%{_sysconfdir}/rdma/)
```

Reply:
That seems like it could work, do you mean it does?

Reply:
Solution A:
```
 # ibacm
 %if 0%{?fedora} >= 33
 %{buildroot}%{_bindir}/ib_acme -D . -O
 %else
 bin/ib_acme -D . -O
 install -D -m0644 ibacm_opts.cfg %{buildroot}%{_sysconfdir}/rdma/
```

Solution B:
```
# ibacm
(if [ -d %{__cmake_builddir} ]; then cd %{__cmake_builddir}; fi
 ./bin/ib_acme -D . -O &&
 install -D -m0644 ibacm_opts.cfg %{buildroot}%{_sysconfdir}/rdma/)
```

Both A and B works for me. Which one do you want?